### ISSUE TITLE
Reviewers using an anonymous private URL to view an anonymized draft dataset can find the dataset author's name in dataset page HTML

### ISSUE THREAD

**Issue by jggautier** [IQSS/dataverse#8158]
**What steps does it take to reproduce the issue?**
In a Dataverse instance with anonymous private URL enabled and configured to hide author name metadata:
- Create a draft dataset version
- Enable an anonymous private URL
- Use that URL to visit the anonymized view of the dataset
- View the HTML for the dataset page and see that the meta tags show the author name

**When does this issue occur?**
When anyone viewing an anonymous private URL looks at the page's HTML

**Which page(s) does it occurs on?**
The dataset page

**What happens?**
If you view the page's HTML, you can see the author name metadata in the meta tags, specifically `<meta name="DC.creator">` and `<meta property="article:author">`

**To whom does it occur (all users, curators, superusers)?**
All users using an anonymous private URL to view an anonymized version of the dataset, but it's problematic when someone reviewing a dataset should not be able to discover who the dataset author is.

**What did you expect to happen?**
The metadata would not be in the page's HTML.

Is it possible to hide that metadata when viewing the source code of an anonymized URL dataset page? Or could all meta tags be removed entirely either when viewing the source code of an anonymized URL dataset page or when viewing an unpublished dataset version? This would mirror how the Schema.org metadata is not in the HTML of unpublished dataset versions.

**Which version of Dataverse are you using?**
5.7

**Any related open or closed issues to this bug report?**
https://github.com/IQSS/dataverse/issues/1724

**Screenshots:**

<kbd><img width="759" alt="Screen Shot 2021-10-14 at 3 17 45 PM" src="https://user-images.githubusercontent.com/18374574/137382381-476887fc-221f-4539-936e-7628692ced91.png"></kbd>

**Comment by qqmyers** [comment]
Good catch - I'd be for removing all the metadata - since this isn't a static/public page, there's no reason for any of it.
